### PR TITLE
feat: Avoid trying to extract PEP-224 docstrings from builtins

### PR DESCRIPTION
**Pull request by Hywan**
If a module or a class is a builtin (e.g. from an extension), the
`_pep224_docstrings` function will generate a lot of warnings. This
patch skips the extraction of the PEP-224 docstrings for builtins
modules or classes.

### CODE REVIEW

File: pdoc/__init__.py [hunk 1]
```diff
@@ -242,8 +242,10 @@ def _pep224_docstrings(doc_obj: Union['Module', 'Class'], *,
     The second dict contains instance variables and is non-empty only in case
     `doc_obj` is a `pdoc.Class` which has `__init__` method.
     """
-    # No variables in namespace packages
-    if isinstance(doc_obj, Module) and doc_obj.is_namespace:
+    is_builtins = type(doc_obj.obj).__module__ == 'builtins'
```
Comment (kernc) [inline pdoc/__init__.py:245]: The tests fail because not all objects have `.__module__` attribute. Use `getattr()`?

**Review by kernc — CHANGES_REQUESTED** [review]
For the record, can you show an example project where "`_pep224_docstrings` function will generate a lot of warnings" on builtin types?

**Comment by Hywan** [comment]
> For the record, can you show an example project where "_pep224_docstrings function will generate a lot of warnings" on builtin types?

The wasmer-python project (https://github.com/wasmerio/wasmer-python/). For every module, class and function, there is a warning :-).

**Comment by Hywan** [comment]
2 tests are failing:

```
======================================================================
FAIL: test_builtin_methoddescriptors (pdoc.test.ApiTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/hywan/Development/pdoc/pdoc/test/__init__.py", line 521, in test_builtin_methoddescriptors
    c = pdoc.Class('STType', pdoc.Module(parser), parser.STType)
AssertionError: UserWarning not triggered

======================================================================
FAIL: test_class_variables_docstring_not_from_obj (pdoc.test.ApiTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/hywan/Development/pdoc/pdoc/test/__init__.py", line 514, in test_class_variables_docstring_not_from_obj
    doc = pdoc.Class('C', pdoc.Module('pdoc'), C)
AssertionError: UserWarning not triggered

----------------------------------------------------------------------
```

Do you think they are still relevant?

**Comment by kernc** [comment]
Surely relevant, but maybe dispose of their `assertWarns` context. `test_builtin_methoddescriptors` was somewhat erroneous (thanks!), so I dared to appropriate it in https://github.com/pdoc3/pdoc/commit/e9fd89f9a389c48d4252559b52b1017788446a7e. Maybe pull/rebase on master.

File: pdoc/__init__.py [hunk 2]
```diff
@@ -242,7 +242,9 @@ def _pep224_docstrings(doc_obj: Union['Module', 'Class'], *,
     The second dict contains instance variables and is non-empty only in case
     `doc_obj` is a `pdoc.Class` which has `__init__` method.
     """
-    # No variables in namespace packages
+    is_builtins = type(doc_obj.obj).__module__ == 'builtins'
```
Comment (kernc) [inline pdoc/__init__.py:245]: Let's move this below to where it's used, then.

**Review by kernc — APPROVED** [review]

**Comment by kernc** [comment]
Merged with slight modification. Thanks!

**Pull request merged**